### ISSUE TITLE
Config flag for disabling `Prefer: count=exact`

### ISSUE THREAD

**Issue by steve-chavez** [PostgREST/postgrest#2777]
We currently rely on this guide for protecting against full counts on big tables: https://postgrest.org/en/v10.2/admin.html#count-header-dos

Which is incomplete because AFAICT Nginx can't remove a header based on a value OOTB. It will always remove the whole header. We'd need OpenResty for this.

Now, this problem has already been solved with the [estimated](https://postgrest.org/en/v10.2/api.html#estimated-count) and [planned](https://postgrest.org/en/v10.2/api.html#planned-count) count.

We just need to disable the `Prefer: count=exact`. We can do this with a config option.

```
db-exact-count-enable = false
```

**Comment by steve-chavez** [comment]
It was brought to my attention that `count=planned` (EXPLAIN) count disregards RLS, hence "leaking" true count in a multi-tenant setup. Is that true?

If so, we'd also need an option to disable `planned` and `estimated`.

**Comment by wolfgangwalther** [comment]
> It was brought to my attention that `count=planned` (EXPLAIN) count disregards RLS, hence "leaking" true count in a multi-tenant setup. Is that true?

I don't think so. In `postgrest-with-pg-18 psql -U postgres`, I run the following:

```
postgres=# CREATE TABLE t AS SELECT generate_series(1,10000) AS c;
SELECT 10000
postgres=# ANALYZE t;
ANALYZE
postgres=# EXPLAIN SELECT * FROM t WHERE c > 5000;
                      QUERY PLAN
------------------------------------------------------
 Seq Scan on t  (cost=0.00..189.00 rows=5000 width=4)
   Filter: (c > 5000)
(2 rows)
```

The first result is a plain explain with the superuser role without any restrictions - after analyze, the expectation is 5000 rows, which is very accurate.

```
postgres=# ALTER TABLE t ENABLE ROW LEVEL SECURITY;
ALTER TABLE
postgres=# CREATE ROLE alice;
CREATE ROLE
postgres=# GRANT SELECT ON TABLE t TO ALICE;
GRANT
postgres=# CREATE POLICY read ON t AS PERMISSIVE FOR SELECT TO alice USING (c > 5000);
CREATE POLICY
postgres=# SET ROLE alice;
SET
postgres=> EXPLAIN SELECT * FROM t;
                      QUERY PLAN
------------------------------------------------------
 Seq Scan on t  (cost=0.00..189.00 rows=5000 width=4)
   Filter: (c > 5000)
(2 rows)
```

Enabling RLS shows two things: The filter from the policy is still visible, catalog data is not private. Thus, the expected number of rows is still correct.

```
postgres=> EXPLAIN SELECT * FROM t WHERE c < 5000;
                    QUERY PLAN
---------------------------------------------------
 Seq Scan on t  (cost=0.00..214.00 rows=1 width=4)
   Filter: ((c > 5000) AND (c < 5000))
(2 rows)
```

Trying to see how many rows "the other" tenant has doesn't work - it returns an estimation of a single row, which is clearly incorrect.
